app/src/androidTest/java/com/github/bgabriel998/softwaredevproject/FullGalleryActivityTest.java

```diff
@@ -48,29 +60,21 @@ public void cleanUp(){
         Intents.release();
     }
 
-    /* Test that the toolbar title is set as expected */
-    @Test
-    public void TestToolbarTitle(){
-        String TOOLBAR_TITLE = "Gallery";
-        ViewInteraction greetingText = Espresso.onView(withId(R.id.toolbarTitle));
-        greetingText.check(matches(withText(TOOLBAR_TITLE)));
-    }
-
-    /* Test that the activity finishes when the toolbar back button is pressed. */
+    /* Test if that the gallery empty text is visible and contains correct text, when gallery is empty */
     @Test
-    public void TestToolbarBackButton(){
-        onView(withId(R.id.toolbarBackButton)).perform(click());
-        assertSame(testRule.getScenario().getResult().getResultCode(), Activity.RESULT_CANCELED);
+    public void TestGalleryEmptyGone(){
+        onView(withId(R.id.gallery_empty)).check(matches(withEffectiveVisibility(ViewMatchers.Visibility.GONE)));
     }
 
     /* Test to scroll trough elements in recycler view and press last item, to see correct intent is sent */
     @Test
     public void TestPressLastItem(){
-        // TODO Redo test when actually using database.
+        String path = CameraActivity.getOutputDirectory(ApplicationProvider.getApplicationContext()) +
+                      "/TestImage19";
         onView(withId(R.id.gallery_recyclerview))
                 .perform(RecyclerViewActions.actionOnItemAtPosition(19,
                         click()));
         intended(allOf(IntentMatchers.hasComponent(ImageActivity.class.getName()),
-                IntentMatchers.hasExtra("image", R.drawable.temp_camera)));
+                IntentMatchers.hasExtra("imagePath", path)));
```

Comment:
```suggestion
                IntentMatchers.hasExtra(ImageActivity.IMAGE_PATH_INTENT, path)));
```
Found the string here too!